test/hotspot/jtreg/compiler/c2/aarch64/TestFloat16Replicate.java

```diff
@@ -0,0 +1,124 @@
+/* Copyright (c) 2025, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @bug 8361582
+* @summary Ensure the correct backend replicate node is being generated for
+*          half precision float constants on >16B SVE machines
+* @modules jdk.incubator.vector
+* @library /test/lib /
+* @run main/othervm compiler.c2.aarch64.TestFloat16Replicate
+*/
+
+package compiler.c2.aarch64;
+
+import compiler.lib.ir_framework.*;
+import java.util.Arrays;
+import jdk.incubator.vector.Float16;
+import jdk.test.lib.*;
+
+import static java.lang.Float.*;
+import static jdk.incubator.vector.Float16.*;
+
+public class TestFloat16Replicate {
+    private static short[] input;
+    private static short[] output;
+
+    // Choose FP16_IN_RANGE which is within the range of [-128 << 8, 127 << 8] and a multiple of 256
+    private static final Float16 FP16_IN_RANGE = Float16.shortBitsToFloat16((short)512);
+
+    // Choose a value out of the range of [-128 << 8, 127 << 8] or a non multiple of 256 for FP16_OUT_OF_RANGE
+    private static final Float16 FP16_OUT_OF_RANGE = Float16.shortBitsToFloat16((short)1035);
+
+    private static final int LEN = 1024;
+
+    public static void main(String args[]) {
+        TestFramework.runWithFlags("--add-modules=jdk.incubator.vector", "-XX:-TieredCompilation");
```

Comment:
What about a run that runs with TieredCompilation? Would be nice to test other modes as well.

Reply:
Hi, thanks for your review. I tried but it doesn't trigger c2 compilation (maybe I need to increase `Warmup`?). As my main motivation was to test the generated backend mach nodes from c2 compilation, I run it specifically with `-XX:-TieredCompilation.`

Reply:
It fails to match the IR nodes. I think it happened when I used a smaller `Warmup`. With the `Warmup` I am using, it seems to be working fine. I will add that case as well.

Reply:
Ok, so then it is probably a profile issue. Thanks for adding both runs!